File: javascript/conduits/transaction.js
export class Transactions {
    static transactionsAwaitingApproval = [];

    static add(txn) {
        this.transactionsAwaitingApproval.push(txn);
    }

    static remove(txn) {
        this.transactionsAwaitingApproval = this.transactionsAwaitingApproval.filter(
            tx => JSON.stringify(tx) !== JSON.stringify(txn)
        );
    }

    static getPendingTransactions() {
        return [...this.transactionsAwaitingApproval];
    }
}

File: rust/examples/conduits/transaction.rs
use std::sync::{Mutex, Arc};
use serde_json::Value;
use lazy_static::lazy_static;

lazy_static! {
    static ref TRANSACTIONS: Arc<Mutex<Vec<Value>>> = Arc::new(Mutex::new(Vec::new()));
}

pub struct Transactions;

impl Transactions {
    pub fn new() -> Self {
        Transactions
    }

    pub fn add(txn: Value) {
        let mut txns = TRANSACTIONS.lock().unwrap();
        txns.push(txn);
    }

    pub fn remove(txn: &Value) {
        let mut txns = TRANSACTIONS.lock().unwrap();
        *txns = txns.iter()
            .filter(|&tx| tx != txn)
            .cloned()
            .collect();
    }

    pub fn get_pending_transactions() -> Vec<Value> {
        let txns = TRANSACTIONS.lock().unwrap();
        txns.clone()
    }
}